app/controllers/actions/AllowAccessAction.scala

```diff
@@ -129,23 +129,50 @@ class AllowAccessAction(
 
 class AllowAccessActionForIdentifierRequest(
                          frontendAppConfig: FrontendAppConfig,
-                         minimalConnector: MinimalConnector
+                         minimalConnector: MinimalConnector,
+                         schemeDetailsConnector: SchemeDetailsConnector,
+                         errorHandler: ErrorHandler,
+                         srnOpt: Option[String]
                        )(
                          implicit val executionContext: ExecutionContext
                        )
   extends ActionFilter[IdentifierRequest] with AllowAccessCommon {
+
+  private def getIdType[A](request: IdentifierRequest[A]):String = {
+    (request.psaId, request.pspId) match {
+      case (Some(_), _) => "psaId"
+      case (_, Some(_)) => "pspId"
+      case _ => throw new Exception("Unable to get ID from request")
+    }
+  }
   override protected def filter[A](request: IdentifierRequest[A]): Future[Option[Result]] = {
     implicit val hc: HeaderCarrier =
       HeaderCarrierConverter.fromRequestAndSession(request, request.session)
-    minimalConnector.getMinimalDetails(implicitly, implicitly, request).map { minimalDetails =>
-      minimalFlagsRedirect(MinimalFlags(minimalDetails.deceasedFlag, minimalDetails.rlsFlag),
-        frontendAppConfig, request.schemeAdministratorType) match {
-        case optionRedirectUrl@Some(_) => optionRedirectUrl
-        case _ => None
+
+    def optFtrToFtrOpt[T](x: Option[Future[T]]): Future[Option[T]] =
+      x match {
+        case Some(f) => f.map(Some(_))
+        case None    => Future.successful(None)
+      }
+
+    //TODO: Potentially add code for both associations
```

Comment:
After further investigation it seems that both enrolments are not needed, you select which enrolment to use when using two enrolments during initial login. Pushed a separate branch just in case with this done.